runner.py

```diff
@@ -1,12 +1,61 @@
 """Script that is able to start Uvicorn-based REST API service."""
 
+import logging
 import os
+from pathlib import Path
 
 import uvicorn
 
+from ols.utils import config
+from ols.utils.logging import configure_logging
+
 if __name__ == "__main__":
     # TODO syedriko consolidate this env var into a central location as per OLS-345.
     # disable Gradio analytics, which calls home to https://api.gradio.app
     os.environ["GRADIO_ANALYTICS_ENABLED"] = "false"
 
-    uvicorn.run("ols.app.main:app", host="127.0.0.1", port=8080, reload=True)
+    cfg_file = os.environ.get("OLS_CONFIG_FILE", "olsconfig.yaml")
+    config.init_config(cfg_file)
+
+    configure_logging(config.ols_config.logging_config)
+    logger = logging.getLogger(__name__)
+    logger.info(f"Config loaded from {Path(cfg_file).resolve()}")
+
+    config.init_query_filter()
+    config.init_vector_index()
+
+    # ruff: noqa: S104
+    host = "127.0.0.1" if config.dev_config.run_on_localhost else "0.0.0.0"
+
+    if config.dev_config.tls_disabled:
+        # TLS is disabled, run without SSL configuration
+        uvicorn.run("ols.app.main:app", host=host, port=8080, workers=1)
```

Comment:
I'd probably add a comment why `workers=1` does the trick

Reply:
resolved